lumibot/tools/polygon_helper.py

```diff
@@ -336,17 +346,30 @@ def load_cache(cache_file):
     return df_feather
 
 
-def update_cache(cache_file, df_all, df_feather):
+def update_cache(cache_file, df_all, missing_dates=None):
     """Update the cache file with the new data"""
     # Check if df_all is different from df_feather (if df_feather exists)
     if df_all is not None and len(df_all) > 0:
-        # Check if the dataframes are the same
-        if df_all.equals(df_feather):
-            return
-
         # Create the directory if it doesn't exist
         cache_file.parent.mkdir(parents=True, exist_ok=True)
 
+        if missing_dates:
+            missing_df = pd.DataFrame(
+                [datetime(year=d.year, month=d.month, day=d.day, tzinfo=LUMIBOT_DEFAULT_PYTZ) for d in missing_dates],
+                columns=["datetime"])
+            missing_df.set_index("datetime", inplace=True)
+            # Set the timezone to UTC
+            missing_df.index = missing_df.index.tz_convert("UTC")
+            df_concat = pd.concat([df_all, missing_df]).sort_index()
+            # Let's be careful and check for duplicates to avoid corrupting the feather file.
+            if df_concat.index.duplicated().any():
+                logging.warn(f"Duplicate index entries found when trying to update Polygon cache {cache_file}")
```

Comment:
These logging messages should be "debug" as there really isn't anything the end user can do about it.  It's actually not really clear what a Developer should do about it either 😀 

Why not simply "drop_duplicates()" in this case?  The "missing_df" rows are added 2nd, so they should be the ones to get dropped anyway.

Reply:
Well, it would indicate a bug and hopefully the warning gets reported and I would look into it.  Being cautious here because I don't have a lot of tests to make me very confident.  

The case that isn't addressed here and hasn't (yet) shown up in my backtests is when df_all is None or empty and missing_dates isn't.  That is actually to be expected if there are no bars for the whole date range requested.  I think what should happen in that case is writing empty bars for all those dates but again not feeling all that brave here (yet ;-).

Reply:
@fovi-com I think this change is a very good idea. I would actually take it a step further and return a df with NaN values where polygon couldn't find any data. That way when you run the back test the second time it won't try to find data for the date range again because it will know that polygon does not have the data. This would significantly improve the performance of options Backtesting. 

I've wanted to make this change for weeks but haven't had the time. If you want to take a stab at it I can review the code to make sure it works properly. No need to worry if you'll break something, we have a bunch of unit tests around this and 2 people to review it.

Reply:
I've gone ahead and changed the empty row logic to work even when all dates are missing.  Also added tests.

Reply:
awsesome! great work @fovi-com !